src/components/Verse/SaveToCollectionAction.tsx

```diff
@@ -0,0 +1,179 @@
+/* eslint-disable max-lines */
+/* eslint-disable react-func/max-lines-per-function */
+import { useState } from 'react';
+
+import useTranslation from 'next-translate/useTranslation';
+import { shallowEqual, useSelector } from 'react-redux';
+import useSWR, { useSWRConfig } from 'swr';
+import useSWRImmutable from 'swr/immutable';
+
+import PlusIcon from '../../../public/icons/plus.svg';
+import SaveToCollectionModal, {
+  Collection,
+} from '../Collection/SaveToCollectionModal/SaveToCollectionModal';
+import PopoverMenu from '../dls/PopoverMenu/PopoverMenu';
+
+import { ToastStatus, useToast } from 'src/components/dls/Toast/Toast';
+import { selectQuranReaderStyles } from 'src/redux/slices/QuranReader/styles';
+import { getMushafId } from 'src/utils/api';
+import {
+  addCollection,
+  addOrRemoveBookmark,
+  getBookmarkCollections,
+  getCollectionsList,
+} from 'src/utils/auth/api';
+import {
+  makeBookmarkCollectionsUrl,
+  makeCollectionsUrl,
+  makeIsResourceBookmarkedUrl,
+} from 'src/utils/auth/apiPaths';
+import { isLoggedIn } from 'src/utils/auth/login';
+import { logButtonClick } from 'src/utils/eventLogger';
+import BookmarkType from 'types/BookmarkType';
+
+const SaveToCollectionAction = ({ verse, bookmarksRangeUrl }) => {
+  const [isSaveCollectionModalOpen, setIsSaveCollectionModalOpen] = useState(false);
+  const quranReaderStyles = useSelector(selectQuranReaderStyles, shallowEqual);
+  const mushafId = getMushafId(quranReaderStyles.quranFont, quranReaderStyles.mushafLines).mushaf;
+  const { t } = useTranslation();
+  const { data: collectionListData, mutate: mutateCollectionListData } = useSWR(
```

Comment:
I noticed we are using `useSWR` here but for `bookmarkCollectionIdsData` we are using `useSWRImmutable`. Is there a reason for that? Also for `bookmarkCollectionIdsData` we check first if the user is logged in vs here we don't